haskell-src/Concordium/Constants.hs

```diff
@@ -25,8 +25,22 @@ maxPayloadSize SP5 = maxPayloadSize SP4
 -- * Web assembly related constants
 
 -- |Maximum length of the parameter to init and receive methods.
-maxParameterLen :: Word16
-maxParameterLen = 1024
+-- The limit was changed from `1024` to `65535` in P5.
+maxParameterLen :: SProtocolVersion pv -> Word16
+maxParameterLen SP1 = 1024
+maxParameterLen SP2 = 1024
+maxParameterLen SP3 = 1024
+maxParameterLen SP4 = 1024
+maxParameterLen _ = 65535
```

Comment:
```suggestion
maxParameterLen SP5 = 65535
```
It is more maintainable this way.

Reply:
I actually thought `_` would be more maintainable as we wouldn't need to update this code for future updates unless we change the limits. But I guess we also risk forgetting something with the wildcard ;)

Reply:
It is more maintainable from the perspective of having to make fewer changes if we add a new protocol version. However this is largely a trivial thing to do and the compiler complains.

It is less maintainable from the perspective that now we have an implicit dependency on the protocol version that the compiler will not complain about.